src/spec.c

```diff
@@ -1799,25 +1856,23 @@ static IndexesScanner *IndexesScanner_New(IndexSpec *spec) {
 void IndexesScanner_Free(IndexesScanner *scanner) {
   if (global_spec_scanner == scanner) {
     global_spec_scanner = NULL;
-  } else if (!scanner->cancelled) {
-    if (scanner->spec && scanner->spec->scanner == scanner) {
-      scanner->spec->scanner = NULL;
-      scanner->spec->scan_in_progress = false;
+  } else {
+    StrongRef tmp = WeakRef_Promote(scanner->spec_ref);
+    IndexSpec *spec = StrongRef_Get(tmp);
+    if (spec) {
+      if (spec->scanner == scanner) {
```

Comment:
I guess this is not needed anymore right?

Reply:
We can set the scanner to NULL without this check no?

Reply:
But what if it was replaced and now we free the irrelevant scanner? It should not edit the spec in this case